packages/video_player/video_player_android/test/android_video_player_test.dart

```diff
@@ -234,16 +234,16 @@ void main() {
     });
 
     test('videoEventsFor', () async {
-      _ambiguate(ServicesBinding.instance)
-          ?.defaultBinaryMessenger
+      _ambiguate(TestDefaultBinaryMessengerBinding.instance)!
+          .defaultBinaryMessenger
           .setMockMessageHandler(
         'flutter.io/videoPlayer/videoEvents123',
         (ByteData? message) async {
           final MethodCall methodCall =
               const StandardMethodCodec().decodeMethodCall(message);
           if (methodCall.method == 'listen') {
-            await _ambiguate(ServicesBinding.instance)
-                ?.defaultBinaryMessenger
+            await _ambiguate(TestDefaultBinaryMessengerBinding.instance)!
+                .defaultBinaryMessenger
                 .handlePlatformMessage(
                     'flutter.io/videoPlayer/videoEvents123',
                     const StandardMethodCodec()
```

Comment:
I wonder why these couple of tests are using `.setMockMessageHandler` instead of `.setMockMethodCallHandler`... Is it so this can use a custom codec to decode the method call line 243?

Reply:
no idea.